src/lockscreen.js

```diff
@@ -193,16 +193,36 @@ var Lockscreen = GObject.registerClass({
         this._activeSoundID = null;
     }
 
-    _updateLockStateWithLock() {
+    _updateLockStateWithLock(retries = null) {
         this._playbin.hasLock = !!this._lock;
 
         if (!this._lock)
             return;
 
         this._updateBackground();
 
-        if (!this._manager.isUnlocked(this._lock))
+        try {
+            if (!this._manager.isUnlocked(this._lock))
+                return;
+        } catch (e) {
+            const retry = retries === null ? 3 : retries;
```

Comment:
This wouldn't be needed if the `retries` was `3` instead of `null` by default (and the code would be more readable too).

Reply:
I've just created a new PR to do that way:

https://github.com/endlessm/hack-toolbox-app/pull/184

Reply:
> Yes, I didn't noticed that

No worried!

> I've just created a new PR to do that way:

Thank you! I'll take a look.